MapboxDirections/MBRouteLeg.swift

```diff
@@ -7,16 +7,19 @@ import Polyline
  */
 @objc(MBRouteLeg)
 open class RouteLeg: NSObject, NSSecureCoding {
+    
+    typealias SegmentAttributes = [SegmentAttribute:[Any]]
     // MARK: Creating a Leg
     
-    internal init(steps: [RouteStep], json: JSONDictionary, source: Waypoint, destination: Waypoint, profileIdentifier: MBDirectionsProfileIdentifier) {
+    internal init(steps: [RouteStep], json: JSONDictionary, source: Waypoint, destination: Waypoint, profileIdentifier: MBDirectionsProfileIdentifier, segmentAttributes: SegmentAttributes? = nil) {
         self.source = source
         self.destination = destination
         self.profileIdentifier = profileIdentifier
         self.steps = steps
         distance = json["distance"] as! Double
         expectedTravelTime = json["duration"] as! Double
         self.name = json["summary"] as! String
+        self.segmentAttributes = segmentAttributes
```

Comment:
There should be two separate properties: `nodeAttributes` and `segmentAttributes`. `nodeAttributes` should contain a `openStreetMapNodeIdentifier` key, and `segmentAttributes` should contain the other keys.